File: backend/src/infrastructure/mock/MockModerationRepository.ts
import type { ContentReport, ReportTargetType } from '../../domain/entities/moderation'
import type { IModerationRepository } from '../../domain/repositories/IModerationRepository'

const reports: ContentReport[] = []
const blocks = new Map<string, Set<string>>()

export class MockModerationRepository implements IModerationRepository {
  async createReport(input: {
    reporterId: string
    targetType: ReportTargetType
    targetId: string
    reason: string
    detail?: string | null
  }): Promise<ContentReport> {
    const report: ContentReport = {
      id: crypto.randomUUID(),
      reporterId: input.reporterId,
      targetType: input.targetType,
      targetId: input.targetId,
      reason: input.reason,
      detail: input.detail ?? null,
      createdAt: new Date().toISOString(),
    }
    reports.push(report)
    return report
  }

  async blockUser(blockerId: string, blockedId: string): Promise<void> {
    const set = blocks.get(blockerId) ?? new Set<string>()
    set.add(blockedId)
    blocks.set(blockerId, set)
  }

  async unblockUser(blockerId: string, blockedId: string): Promise<void> {
    blocks.get(blockerId)?.delete(blockedId)
  }

  async listBlockedUserIds(blockerId: string): Promise<string[]> {
    return [...(blocks.get(blockerId) ?? [])]
  }
}

File: lib/infrastructure/mock/mock_moderation_repository.dart
import '../moderation_repository.dart';

class MockModerationRepository implements ModerationRepository {
  final Set<String> _blockedUserIds = {};

  @override
  Future<void> reportContent({
    required ReportTargetType targetType,
    required String targetId,
    required ContentReportReason reason,
    String? detail,
  }) async {}

  @override
  Future<void> blockUser(String userId) async {
    _blockedUserIds.add(userId);
  }

  @override
  Future<void> unblockUser(String userId) async {
    _blockedUserIds.remove(userId);
  }

  @override
  Future<Set<String>> getBlockedUserIds() async {
    return Set.unmodifiable(_blockedUserIds);
  }
}
